kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -57,6 +59,16 @@ private[kyuubi] class EngineRef private(conf: KyuubiConf, user: String, sessionI
   private val shareLevel: ShareLevel = ShareLevel.withName(conf.get(ENGINE_SHARE_LEVEL))
 
   private val subDomain: Option[String] = conf.get(ENGINE_SHARE_LEVEL_SUB_DOMAIN)
+    .orElse {
+      if (conf.get(ENGINE_POOL_ENABLED)) {
```

Comment:
how about we save a config here?  if the final poolSize > 1

Reply:
hi, you mean remove `ENGINE_POOL_ENABLED` and use `poolSize > 1` to determine whether to enable engine pool?

Reply:
yes